src/Controller/AbstractController.php

```diff
@@ -196,35 +220,25 @@ protected function createSupportedJsonResponse($data, $statusCode = 200)
         return $response;
     }
 
-    /**
-     * Get the master request.
-     *
-     * @return Request
-     */
-    protected function getRequest()
+    protected function getRequest(): Request
     {
-        if (version_compare(Kernel::VERSION, '2.4', '<=')) {
-            return $this->container->get('request');
-        }
+        /** @var RequestStack $requestStack */
+        $requestStack = $this->container->get('request_stack');
+
+        /** @var Request $request */
+        $request = $requestStack->getMasterRequest();
 
-        return $this->container->get('request_stack')->getMasterRequest();
+        return $request;
     }
 
     /**
      * Event dispatch proxy that avoids using deprecated interfaces.
-     *
-     * @param $event
      */
-    protected function dispatchEvent($event, string $eventName): void
+    protected function dispatchEvent(Event $event, string $eventName = null): void
     {
+        /** @var EventDispatcherInterface $dispatcher */
         $dispatcher = $this->container->get('event_dispatcher');
 
-        if ($dispatcher instanceof EventDispatcherInterface) {
-            $dispatcher->dispatch($event, $eventName);
-            $dispatcher->dispatch($event, sprintf('%s.%s', $eventName, $this->type));
-        } else {
-            $dispatcher->dispatch($eventName, $event);
-            $dispatcher->dispatch(sprintf('%s.%s', $eventName, $this->type), $event);
-        }
+        $dispatcher->dispatch($event, $eventName);
```

Comment:
Shouldn't this be like this?

```php
            $dispatcher->dispatch($event, $eventName);
            $dispatcher->dispatch($event, sprintf('%s.%s', $eventName, $this->type));
```

Currently the scoped events aka `UploadEvents::postPersist($mapping)` are not working at all.

Reply:
Good catch! I'll have a look. 🙃

Reply:
This is probably missing as well in `src/Controller/AbstractChunkedController.php`.

Reply:
Fixed in 8526d25.